author	David Herrmann <dh.herrmann@googlemail.com>
commit	dfe14a340f8135c28d41be20b03ce21d7aa7e3d9
	Thu, 15 Mar 2012 13:34:03 +0000 (10:34 -0300)

wiimote: Detect Wii Remotes by device name

We currently detect Wii Remotes by DID information. However, not all Wii
Remotes (especially 3rd party Wii Remotes) provide DID information.
Moreover, they provide no way of identifying them aside from their
device name. Hence, this adds an experimental name-detection for Wii
Remotes.

diff -cr bluez-4.99.pan.orig/plugins/wiimote.c bluez-4.99.pan/plugins/wiimote.c
*** bluez-4.99.pan.orig/plugins/wiimote.c	Wed Jan 11 17:44:59 2012
--- bluez-4.99.pan/plugins/wiimote.c	Fri Oct 12 08:09:38 2012
***************
*** 26,31 ****
--- 26,32 ----
  #endif
  
  #include <bluetooth/bluetooth.h>
+ #include <glib.h>
  
  #include "plugin.h"
  #include "adapter.h"
***************
*** 60,77 ****
  {
  	uint16_t vendor, product;
  	bdaddr_t sba, dba;
! 	char addr[18];
  
  	adapter_get_address(adapter, &sba);
  	device_get_address(device, &dba, NULL);
  	ba2str(&dba, addr);
  
  	vendor = btd_device_get_vendor(device);
- 	if (vendor != 0x057e)
- 		return 0;
  
  	product = btd_device_get_product(device);
! 	if (product == 0x0306) {
  		DBG("Forcing fixed pin on detected wiimote %s", addr);
  		memcpy(pinbuf, &sba, 6);
  		return 6;
--- 61,81 ----
  {
  	uint16_t vendor, product;
  	bdaddr_t sba, dba;
! 	char addr[18], name[25];
  
  	adapter_get_address(adapter, &sba);
  	device_get_address(device, &dba, NULL);
  	ba2str(&dba, addr);
  
  	vendor = btd_device_get_vendor(device);
  
  	product = btd_device_get_product(device);
! 
! 	device_get_name(device, name, sizeof(name));
! 	name[sizeof(name) - 1] = 0;
! 
! 	if (g_str_equal(name, "Nintendo RVL-CNT-01") ||
! 				(vendor == 0x057e && product == 0x0306)) {
  		DBG("Forcing fixed pin on detected wiimote %s", addr);
  		memcpy(pinbuf, &sba, 6);
  		return 6;
